nixos/modules/system/boot/networkd.nix

```diff
@@ -1192,6 +1193,10 @@ in
         after = [ "systemd-udev-settle.service" ];
       };
 
+      systemd.sockets.systemd-networkd = {
+        wantedBy = [ "sockets.target" ];
+      };
+
```

Comment:
dbus activation is missing from this PR - see https://github.com/NixOS/nixpkgs/pull/88158/files#diff-7d08c03fc593d925ed022e395579c866R1193.

Please also cherry-pick the timesyncd and resolved commits from #88158.

Reply:
As per https://github.com/NixOS/nixpkgs/pull/89064#issuecomment-636753727, let's do this in #88158.